gaj23/fitlit-starter-kit
# Gaj/feature/iteration 2 hydration data

### What does this PR do? 
>Clearly and concisely describe the feature.  
This feature completes the data portion of Iteration 2: 
- includes test file
- includes original data for testing
- includes Hydration class that passes HAPPY path testing
- Refactoring of code based on eslinter and Turing rules

### Is this a feature or refactor?  
A feature with some syntax refactoring.

### Problems Encountered & Resulting Solutions  
>State any problems encountered 
>Describe your code changes in detail for reviewers. 
I struggled to conceptualize how we wanted our data to be stored and generated. Thao guided me through a large portion of the code written. I completed the last function on my own by using slice and then mapping the resulting array for the necessary data. 

For developer empathy purposes, I wanted to store a specific user's hydration data as a variable. I struggled to understand why my test wasn't failing. I later realized it was because my test wasn't testing the property that I was assigning data to, but the method that doesn't return any data. (See test 

### Where should the reviewer start?
>What file(s) and line(s) are where changes are made?

Please see Hydration files. 


### How should this be tested?
>What steps need to be taken to confirm functionality? 
Please run `npm test test/Hydration-test.js` in your terminal to confirm all tests are passing.


### Steps To Take
>Any questions and possible steps moving forward.
To DRY up code, I would like to understand how to utilizer a helper file as well as how to test that helper file.
